### ISSUE TITLE
rehype-mathjax for math block

### ISSUE THREAD

**Issue by cryoguy** [rasendubi/uniorg#5]
I can not seem to get mathjax to work? Add the following line in `orgToHtml.js`:

```
import mathjax from 'rehype-mathjax';
...
.use(mathjax)
```
 will generate the following errors.

```
node:internal/fs/utils:322
    throw err;
    ^

Error: ENOENT: no such file or directory, lstat '/home/tan/proj/uniorg/examples/org-braindump/braindump/events'
    at realpathSync (node:fs:1777:7)
```

**Comment by rasendubi** [comment]
Can you post a repository that I can use to reproduce the issue? I have followed the steps and there is no error.

Also, what is `braindump/events` directory?

**Comment by cryoguy** [comment]
There is no `braindump/events` as a folder or file. I will try to come up with an MWE.  It will take some time to dissect over 800 org files.  Without mathjax, everything is working fine.  will report back to you.

**Comment by cryoguy** [comment]
hmm!  This is very bizarre. This is happening even with a fresh pull. As soon as I put `import mathjax from 'rehype-mathjax';` in `orgToHtml.js` (even without `.use(mathjax)`), it will spill out the following error. BTW,  I did confirm that `rehype-mathjax` is working. 

```
yarn dev
yarn run v1.22.10
$ next dev
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
ready - started server on http://localhost:3000
event - compiled successfully
event - build page: /next/dist/pages/_error
wait  - compiling...
event - compiled successfully
event - build page: /[[...slug]]
wait  - compiling...
node:internal/fs/utils:322
    throw err;
    ^

Error: ENOENT: no such file or directory, lstat '/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/events'
    at realpathSync (node:fs:1777:7)
    at handleExternals (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/next/dist/build/webpack-config.js:66:53)
    at webpackConfig.externals (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/next/dist/build/webpack-config.js:78:136)
    at handleExternals (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/ExternalModuleFactoryPlugin.js:78:17)
    at next (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/ExternalModuleFactoryPlugin.js:66:9)
    at handleExternals (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/ExternalModuleFactoryPlugin.js:71:7)
    at /home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/ExternalModuleFactoryPlugin.js:101:5
    at /home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/NormalModuleFactory.js:400:5
    at AsyncSeriesWaterfallHook.eval [as callAsync] (eval at create (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at NormalModuleFactory.create (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/NormalModuleFactory.js:381:28)
    at /home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/Compilation.js:897:14
    at Semaphore.acquire (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/util/Semaphore.js:29:4)
    at asyncLib.forEach.err.stack (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/Compilation.js:895:15)
    at arrayEach (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/neo-async/async.js:2405:9)
    at Object.each (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/neo-async/async.js:2846:9)
    at Compilation.addModuleDependencies (/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/node_modules/webpack/lib/Compilation.js:873:12) {
  errno: -2,
  syscall: 'lstat',
  code: 'ENOENT',
  path: '/home/tan/tmp/testing/uniorg/examples/org-braindump/braindump/events'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by rasendubi** [comment]
This looks like something on your system is creating a temporary `events` file or directory and removes it right after webpack has detected it (and tried to query more info on it). The error clearly comes from webpack and none of the org-braindump code is involved. I'm not sure how I can help you here :confused:

**Comment by cryoguy** [comment]
Thanks for your help, again.  which webpack version you are using?

**Comment by rasendubi** [comment]
webpack-4.44.1

It's encoded in `package-lock.json`. As an option, try using npm instead of yarn. I use node-14.15.4, npm-6.14.10.

**Comment by cryoguy** [comment]
I had no luck with `rehype-mathjax`.  So I switched to `rehype-katex`. This is what I am getting right now:  it did not crashed as `rehype-mathjax` did. But it did  not render the latex fully.  I getting some warnings during compiling:

```
LaTeX-incompatible input and strict mode is set to 'warn': Unrecognized Unicode character "−" (8722) [unknownSymbol]
```
and the following is what I got with your README example.

![uniorg](https://user-images.githubusercontent.com/12690537/107442242-9a172900-6aeb-11eb-9f9d-a011a252b98f.png)

**Comment by rasendubi** [comment]
Did you import katex styles?

```ts
import 'katex/dist/katex.min.css';
```

**Comment by cryoguy** [comment]
I had no idea that I need to import the katex css, and its fonts. It is working now, Thanks!

What do I need to do to render the following?

```
\begin{equation} \begin{bmatrix} 1 & 1 & 1 \\ 3 & -2 & 1 \\ 2 & 1 &-1 \end{bmatrix} \begin{bmatrix} a \\ b \\ c \end{bmatrix}= \begin{bmatrix} 6 & 2 & 1 \end{bmatrix} \end{equation}
```

**Comment by rasendubi** [comment]
katex does not currently support `equation` environment: https://github.com/KaTeX/KaTeX/issues/445

There is a workaround there to add `\begin{equation}`/`\end{equation}` as a custom delimiter to katex. You can pass delimiters as an additional option to rehype-katex: https://github.com/remarkjs/remark-math/tree/main/packages/rehype-katex#options-1

**Comment by cryoguy** [comment]
Adding the following to `orgToHtml.js` did not complain, but did not render either.   any pointer?

```
  .use(katex,{
  delimiters: [
    {left: "$$", right: "$$", display: true},
    {left: "\\(", right: "\\)", display: false},
    {left: "\\[", right: "\\]", display: true},
    {left: "\\begin{equation}", right: "\\end{equation}", display: true}
  ]
})
```

**Comment by rasendubi** [comment]
Well. It looks like `delimiters` is only supported in [Auto-render Extension](https://katex.org/docs/autorender.html) and it doesn't work for rehype-katex.

Another hope is to update katex itself. The `equation` environment support [has landed](https://github.com/KaTeX/KaTeX/pull/2369) there, but no new version were released. You can try overriding `rehype-katex->katex` dependency with yarn to use version from github, but I'm not sure how to do that exactly.

**Comment by cryoguy** [comment]
Thank you so much for the clarification! I will play with it and update here if I reach sth meaningful.

**Comment by rasendubi** [comment]
If someone stumbled upon this issue, the `equation` environment support has been released, and it works for me with rehype-katex@5.0.0 and katex@0.13.5.

**Comment by cryoguy** [comment]
@rasendubi confirmed.  Thanks!!
